tests/integration/test_project_navigator.py

```diff
@@ -0,0 +1,161 @@
+import mock
+from selenium.webdriver.support.wait import WebDriverWait
+from group_project_v2.components.project_navigator import ViewTypes
+from group_project_v2.components.stage import StageState
+from tests.integration.base_test import SingleScenarioTestSuite
+from tests.integration.page_elements import NavigationViewElement, ResourcesViewElement, SubmissionsViewElement
+from tests.utils import KNOWN_USERS
+
+
+class TestProjectNavigatorViews(SingleScenarioTestSuite):
+    scenario = "example_1.xml"
+
+    def _assert_view_visibility(self, project_navigator, available_views, visible_view):
+        hidden_views = set(available_views) - {visible_view}
+        for view in hidden_views:
+            try:
+                self.assertFalse(project_navigator.get_view_by_type(view).is_displayed())
+            except AssertionError as exception:
+                raise AssertionError("View {view_name} should be hidden".format(view_name=view))
+
+        self.assertTrue(project_navigator.get_view_by_type(visible_view).is_displayed())
+
+    def _assert_view_switching(self, project_navigator, available_views, target_view):
```

Comment:
Could use a docstring here. In fact I don't see many docstrings at all-- and while I don't really think you need to have one for all of them, please do a pass to see if you can find instances where it is not obvious what a function is doing and doc it up.